base/src/main/java/org/aya/tyck/ExprTycker.java

```diff
@@ -366,19 +366,20 @@ else if (iter.hasNext()) {
     };
   }
 
-  private TacElabResult elaborateTactic(TacNode tacNode, Term term) {
+  private @NotNull TacElabResult elaborateTactic(TacNode tacNode, Term term) {
     return switch (tacNode) {
       case TacNode.ExprTac exprTac -> new TacElabResult(exprTac.expr(), inherit(exprTac.expr(), term));
       case TacNode.ListExprTac listExprTac -> {
+        TacElabResult result = null;
+
         var tacNodes = listExprTac.tacNodes();
         var headNode = tacNodes.first();
         var tailNodes = tacNodes.drop(1);
+        // enter into a new local state
+        localCtx = localCtx.deriveMap();
```

Comment:
You can create a local of the old `localCtx` variable here, and use it later to avoid `requireNotNull`

Reply:
This is now fixed.

Yeah, there will a lot of more tests added tomorrow.